pkg/manager/compressor/handler/grpc/handler.go

```diff
@@ -0,0 +1,185 @@
+//
+// Copyright (C) 2019 Vdaas.org Vald team ( kpango, kmrmt, rinx )
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// Package grpc provides grpc server logic
+package grpc
+
+import (
+	"context"
+
+	"github.com/vdaas/vald/apis/grpc/manager/compressor"
+	"github.com/vdaas/vald/apis/grpc/payload"
+	"github.com/vdaas/vald/internal/net/grpc/status"
+	"github.com/vdaas/vald/pkg/manager/compressor/service"
+)
+
+type Server compressor.BackupServer
+
+type server struct {
+	backup     service.Backup
+	compressor service.Compressor
+}
+
+type errDetail struct {
+	method string
+	uuid   string
+	uuids  []string
+}
+
+func New(opts ...Option) Server {
+	s := new(server)
+
+	for _, opt := range append(defaultOpts, opts...) {
+		opt(s)
+	}
+	return s
+}
+
+func (s *server) GetVector(ctx context.Context, req *payload.Backup_GetVector_Request) (res *payload.Backup_MetaVector, err error) {
+	r, err := s.backup.GetObject(ctx, req.GetUuid())
+	if err != nil {
+		detail := errDetail{method: "GetVector", uuid: req.Uuid}
+		return nil, status.WrapWithUnknown("Unknown error occurred", &detail, err)
+	}
+
+	vector, err := s.compressor.Decompress(ctx, r.GetVector())
+	if err != nil {
+		detail := errDetail{method: "GetVector", uuid: req.Uuid}
+		return nil, status.WrapWithInternal("Internal error occurred", &detail, err)
+	}
+
+	return &payload.Backup_MetaVector{
+		Uuid:   r.GetUuid(),
+		Meta:   r.GetMeta(),
+		Vector: vector,
+		Ips:    r.GetIps(),
+	}, nil
+}
+
+func (s *server) Locations(ctx context.Context, req *payload.Backup_Locations_Request) (res *payload.Info_IPs, err error) {
+	r, err := s.backup.GetLocation(ctx, req.GetUuid())
+	if err != nil {
+		detail := errDetail{method: "Locations", uuid: req.Uuid}
+		return nil, status.WrapWithUnknown("Unknown error occurred", &detail, err)
+	}
+
+	return &payload.Info_IPs{
+		Ip: r,
+	}, nil
+}
+
+func (s *server) Register(ctx context.Context, meta *payload.Backup_MetaVector) (res *payload.Empty, err error) {
+	vector, err := s.compressor.Compress(ctx, meta.GetVector())
+	if err != nil {
```

Comment:
![badge](https://img.shields.io/badge/review-note-informational.svg?style=flat-square)
There should be fallback method to save plain vector when compress fails.
However, it may be difficult to accomplish it immediately, because the mysql and cassandra column type is binary.
I added that task to the project board.